### PR TITLE
fix: allow nulls in struct

### PR DESCRIPTION
**Pull request by smcquis**
fix: allow nulls in struct for OFAC compliance

### CODE REVIEW

File: cmd/server/download.go [hunk 1]
```diff
@@ -46,26 +46,26 @@ type Download struct {
...
+	SDNs              int `json:"SDNs,omitempty"`
+	Alts              int `json:"altNames,omitempty"`
+	Addresses         int `json:"addresses,omitempty"`
+	SectoralSanctions int `json:"sectoralSanctions,omitempty"`
 
 	// US Bureau of Industry and Security (BIS)
-	DeniedPersons int `json:"deniedPersons"`
-	BISEntities   int `json:"bisEntities"`
+	DeniedPersons int `json:"deniedPersons,omitempty"`
+	BISEntities   int `json:"bisEntities,omitempty"`
```
Comment (elovelan) [inline cmd/server/download.go:68]: when would these ever be `nil`?

File: cmd/server/search_handlers_test.go [hunk 2]
```diff
@@ -313,12 +313,12 @@ func TestSearch__Name(t *testing.T) {
...
-		Alts []*ofac.AlternateIdentity `json:"altNames"`
-		SSIs []*csl.SSI                `json:"sectoralSanctions"`
+		SDNs []*ofac.SDN               `json:"SDNs,omitempty"`
+		Alts []*ofac.AlternateIdentity `json:"altNames,omitempty"`
+		SSIs []*csl.SSI                `json:"sectoralSanctions,omitempty"`
 		// BIS
-		DPs []*dpl.DPL `json:"deniedPersons"`
-		ELs []*csl.EL  `json:"bisEntities"`
+		DPs []*dpl.DPL `json:"deniedPersons,omitempty"`
+		ELs []*csl.EL  `json:"bisEntities,omitempty"`
```
Comment (elovelan) [inline cmd/server/search_handlers_test.go:321]: were the tests failing without these?

**Review by elovelan — COMMENTED** [review]
just some questions about why extra structs were also changed

File: cmd/server/search_handlers_test.go [hunk 3]
```diff
@@ -250,13 +250,13 @@ func TestSearch__NameAndAltName(t *testing.T) {
...
-		SectoralSanctions []*csl.SSI                `json:"sectoralSanctions"`
+		SDNs              []*ofac.SDN               `json:"SDNs,omitempty"`
+		AltNames          []*ofac.AlternateIdentity `json:"altNames,omitempty"`
+		Addresses         []*ofac.Address           `json:"addresses,omitempty"`
+		SectoralSanctions []*csl.SSI                `json:"sectoralSanctions,omitempty"`
 		// BIS
-		DeniedPersons []*dpl.DPL `json:"deniedPersons"`
-		BISEntities   []*csl.EL  `json:"bisEntities"`
+		DeniedPersons []*dpl.DPL `json:"deniedPersons,omitempty"`
+		BISEntities   []*csl.EL  `json:"bisEntities,omitempty"`
```
Comment (elovelan) [inline cmd/server/search_handlers_test.go:259]: were the tests failing without these (same question for entire file)?

**Comment by elovelan** [comment]
Does the OAI spec need to change as well to support these changes?

**Comment by elovelan** [comment]
@smcquis, we should make this change to `main` as well and open a PR to the [original](/moov-io/watchman)

File: cmd/server/download.go (hunk 1 above)
Comment (smcquis) [reply cmd/server/download.go:68]: Ran the job and SDNs, addresses, sectoralsanctions, bisentities had errors in the console output window.  So searched code for those values and saw that most had the omitempty tag on structs but 3 did not.  Changed the structs to match the other strunts and error disappeared.   One of these structs was the test struct but to keep things consistent change that one also.

File: cmd/server/search_handlers_test.go (hunk 3 above)
Comment (smcquis) [reply cmd/server/search_handlers_test.go:259]: Changed 2 other structs to fix null issues.  The test struct was changed to keep things consistent.  Was not failing.  Can undo this and retest if you wish.

File: cmd/server/download.go (hunk 1 above)
Comment (smcquis) [reply cmd/server/download.go:68]: Can change the 2 back and retest.  These 2 are stats structs so would not think they would have issue.

File: cmd/server/search_handlers_test.go (hunk 3 above)
Comment (smcquis) [reply cmd/server/search_handlers_test.go:259]: removed changes and retested.

**Pull request merged**